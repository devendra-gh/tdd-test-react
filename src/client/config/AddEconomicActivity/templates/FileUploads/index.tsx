import React from 'react';
import Form from '@tamm/ui-lib-v2-form';
import { IVariables } from '@tamm/app-composer';
import FileUpload from '@tamm/ui-lib-v2-file-upload';
import InternalApi from 'client/services/InternalApi';
import { has } from 'lodash';

/* istanbul ignore file */

function FileUploads(props: IVariables) {
  const {
    inputGroups,
    handleChange,
    fileUploadData,
    i18n,
    startShowingErrors,
  } = props;
  const { documents } = fileUploadData;

  const onFileUpload = (fieldName: string) => {
    return async (files: IVariables) => {
      try {
        handleChange({
          documents: {
            ...documents,
            [fieldName]: files.map((uploadedFile: any, i: number) => ({
              id: i,
              fieldName,
              lastModifiedDate: uploadedFile.lastModifiedDate,
              name: uploadedFile.name,
              documentName: uploadedFile.name,
              type: uploadedFile.type,
              lastModified: uploadedFile.lastModified,
              size: Math.round(uploadedFile.size / 2),
              loading: true,
            })),
          },
        });

        const documentCategories: IVariables = {
          activitySupportingDoc: 'activitySupportingDoc',
        };
        const promise = files.map(async (uploadedFile: any) => {
          const res = await InternalApi.uploadDedDoc(uploadedFile);
          const payload = {
            fieldName,
            id: res.data.uploadedFileDetails.nameOfFile,
            lastModifiedDate: uploadedFile.lastModifiedDate,
            name: uploadedFile.name,
            documentName: uploadedFile.name,
            documentCategory: documentCategories[fieldName],
            documentPath: res.data.uploadedFileDetails.s3FilePath,
            type: uploadedFile.type,
            lastModified: uploadedFile.lastModified,
            size: uploadedFile.size,
            loading: false,
          };
          return payload;
        });

        const resDocuments = await Promise.all(promise);

        handleChange({
          documents: {
            ...documents,
            [fieldName]: resDocuments,
          },
        });
      } catch (err) {
        handleChange({
          documents: {
            ...documents,
            [fieldName]: null,
          },
        });
        console.error('Error while uploading document:', err);
      }
    };
  };

  const onFileRemove = (fieldName: string) => {
    return () => {
      handleChange({
        documents: {
          ...documents,
          [fieldName]: null,
        },
      });
    };
  };

  const isUploaded = (fieldName: string) => {
    const isUploadedBool = has(documents, `${fieldName}[0].documentPath`);
    return isUploadedBool;
  };

  const getFiles = (fieldName: string) => {
    return documents[fieldName] === null || documents[fieldName] === undefined
      ? []
      : documents[fieldName].map((document: IVariables) => ({
          file: document,
          status: document.loading ? 'progress' : 'success',
          uploaded: document.loading ? 2.2 : document.size,
          loading: document.loading,
        }));
  };

  return (
    inputGroups && (
      <Form name="file-upload-section">
        {inputGroups.map((formGroups: IVariables, index: number) => {
          return (
            <div key={`${index.toString()}`}>
              <Form.Fieldset title={i18n(formGroups.name)}>
                {formGroups.fields.map(
                  (fileConfig: IVariables, indexID: number) => {
                    const {
                      name,
                      label,
                      help,
                      message,
                      ...restProps
                    } = fileConfig;
                    if (name === 'activitySupportingDoc')
                      return (
                        <div key={`${indexID.toString()}`}>
                          <FileUpload
                            i18n={i18n}
                            accept={['application/pdf']}
                            disabled={false}
                            help={
                              (startShowingErrors &&
                                !isUploaded(name) &&
                                i18n(help)) ||
                              (!isUploaded(name) && i18n(message)) ||
                              ''
                            }
                            label={i18n(label)}
                            maxSize={100002200}
                            multiple={false}
                            name={name}
                            validateStatus={
                              startShowingErrors && !isUploaded(name)
                                ? 'error'
                                : ''
                            }
                            {...restProps}
                            files={getFiles(name)}
                            onChange={onFileUpload(name)}
                            onRemove={onFileRemove(name)}
                          />
                        </div>
                      );
                    return null;
                  },
                )}
              </Form.Fieldset>
            </div>
          );
        })}
      </Form>
    )
  );
}
export default FileUploads;
