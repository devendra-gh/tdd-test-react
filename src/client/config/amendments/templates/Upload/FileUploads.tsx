/* eslint-disable complexity */
import React from 'react';
import Form from '@tamm/ui-lib-v2-form';
import { IVariables } from '@tamm/app-composer';
import Grid from '@tamm/ui-lib-v2-grid';
import FileUpload from '@tamm/ui-lib-v2-file-upload';
import InternalApi from 'client/services/InternalApi';
import { AMENDMENT_CATEGORIES as categories } from 'client/config/amendments/constants/amendmentObjects';
import Alert from '@tamm/ui-lib-v2-alert';

const { Row, Col } = Grid;

function FileUploads(props: IVariables) {
  const {
    inputGroups,
    handleChange,
    values: documents,
    i18n,
    locale,
    startShowingErrors,
  } = props;

  const onFileRemove = (fieldName: string) => {
    return () => {
      handleChange(
        documents.filter((i: IVariables) => i.fieldName !== fieldName),
      );
    };
  };

  const handleFileChange = (fieldName: string, files: IVariables[] | null) => {
    if (files === null) {
      handleChange(
        documents.filter((i: IVariables) => i.fieldName !== fieldName),
      );
    } else {
      handleChange([...documents, ...files]);
    }
  };

  const onFileUpload = (fieldName: string) => {
    return async (files: IVariables) => {
      try {
        handleFileChange(
          fieldName,
          files.map((uploadedFile: any, i: number) => ({
            id: i,
            fieldName,
            name: uploadedFile.name,
            type: uploadedFile.type,
            size: Math.round(uploadedFile.size / 2),
            loading: true,
          })),
        );
        const promise = files.map(async (uploadedFile: any) => {
          const res = await InternalApi.uploadDedDoc(uploadedFile);

          const payload = {
            fieldName,
            id: res.data.uploadedFileDetails.nameOfFile,
            name: uploadedFile.name,
            documentPath: res.data.uploadedFileDetails.s3FilePath,
            type: uploadedFile.type,
            size: uploadedFile.size,
          };
          return payload;
        });

        const resDocuments: IVariables[] = await Promise.all(promise);

        handleFileChange(fieldName, resDocuments);
      } catch (err) {
        handleFileChange(fieldName, null);
        console.error('Error while uploading document:', err);
      }
    };
  };

  const getFiles = (fieldName: string) => {
    return (
      documents &&
      documents
        .filter((i: IVariables) => i.fieldName === fieldName)
        .map((document: IVariables) => ({
          file: document,
          status: document.loading ? 'progress' : 'success',
          uploaded: document.loading ? 1 : document.size,
          loading: document.loading || false,
          size: document.size || '',
        }))
    );
  };

  const fileValid = (fieldName: string) => {
    const isValueInRedux = documents.some(
      (file: IVariables) => file.fieldName === fieldName,
    );
    return isValueInRedux ? false : i18n('validationMessage.fileRequired');
  };

  const renderFileSet = (sectionName: string, formFields: IVariables) => {
    return (
      <>
        <Row gutter={20} flex>
          {formFields.map((fileConfig: IVariables) => {
            const { name, description, required, ...restProps } = fileConfig;
            const fileName = `${sectionName}.${name}`;
            const uploadedFiles = getFiles(fileName);
            const label = `${i18n(description || '')}${required ? ' *' : ''}`;

            const fileStatus =
              startShowingErrors && required && fileValid(fileName);
            let isUploading: boolean = false;
            let helpString: string = '';
            if (
              uploadedFiles &&
              uploadedFiles.length &&
              uploadedFiles[0].loading
            ) {
              isUploading = true;
              helpString = i18n('validationMessage.fileUploading');
            } else if (fileStatus) {
              helpString = fileStatus;
            } else {
              helpString = i18n('uploads.fileHelp');
            }

            return (
              <Col xs={12} xl={12} key={fileName}>
                <div id={fileName}>
                  <FileUpload
                    i18n={i18n}
                    aria-label={i18n(`uploadField.${name}`)}
                    label={label}
                    accept={['application/pdf']}
                    disabled={false}
                    maxSize={4e6}
                    // maxSize={100002200}
                    multiple={false}
                    name={fileName}
                    {...restProps}
                    validateStatus={isUploading || fileStatus ? 'error' : ''}
                    files={uploadedFiles}
                    help={helpString}
                    onChange={onFileUpload(fileName)}
                    onRemove={onFileRemove(fileName)}
                  />
                </div>
              </Col>
            );
          })}
        </Row>
        <div className="height-20" />
      </>
    );
  };

  if (!inputGroups || inputGroups.length === 0) {
    return (
      <div id="errorMessage">
        <Alert message={i18n('uploads.error.reqDocsEmpty')} status="warning" />
        <div className="height-40" />
      </div>
    );
  }

  return (
    inputGroups && (
      <div className="file-uploads-wrapper">
        {inputGroups.map((formGroups: IVariables, index: number) => (
          <div key={formGroups.name}>
            <Form.Fieldset
              title={i18n(`uploads.groupTitle.${formGroups.name}`)}
              twoColumns={false}
            >
              {formGroups.description && (
                <p className="subtitleColor">{i18n(formGroups.description)}</p>
              )}
            </Form.Fieldset>
            {index === 0 && <div className="height-20" />}
            {formGroups.name === categories.OWNERSHIP
              ? formGroups.sections.map((formSection: IVariables) => {
                  const fileName = `${formSection.name}.${formSection.referenceKey}`;
                  return (
                    <div key={formSection.name}>
                      <h6>
                        {i18n(`uploads.for.${formSection.name}`)}
                        {` - ${
                          locale === 'en'
                            ? formSection.userEn
                            : formSection.userAr
                        } `}
                        {formSection.action === 'delete' && (
                          <span>({i18n('global.removed')})</span>
                        )}
                      </h6>
                      {renderFileSet(fileName, formSection.fields)}
                    </div>
                  );
                })
              : renderFileSet(formGroups.name, formGroups.sections[0].fields)}
          </div>
        ))}
      </div>
    )
  );
}
export default FileUploads;
