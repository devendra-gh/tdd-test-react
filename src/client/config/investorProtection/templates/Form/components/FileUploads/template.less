.investor-form {
  padding-bottom: 4rem;

  .ui-lib-file-upload {
    max-width: unset !important; // override style from common override file
    &__preview {
      // TODO: need to remove once override file update
      margin-bottom: 0; // dls multiple file upload fix
    }

    &__help {
      width: auto !important;
    }
  }
}
