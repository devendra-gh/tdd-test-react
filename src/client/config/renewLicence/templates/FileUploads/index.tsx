import React from 'react';
import Form from '@tamm/ui-lib-v2-form';
import { IVariables } from '@tamm/app-composer';
import FileUpload from '@tamm/ui-lib-v2-file-upload';
import InternalApi from 'client/services/InternalApi';
import { has } from 'lodash';

function FileUploads(props: IVariables) {
  const {
    inputGroups,
    handleChange,
    fileUploadData,
    i18n,
    startShowingErrors,
  } = props;
  const { documents } = fileUploadData;

  const onFileUpload = (fieldName: string) => {
    return async (files: IVariables) => {
      try {
        handleChange({
          documents: {
            ...documents,
            [fieldName]: files.map((uploadedFile: any, i: number) => ({
              id: i,
              fieldName,
              lastModifiedDate: uploadedFile.lastModifiedDate,
              name: uploadedFile.name,
              documentName: uploadedFile.name,
              type: uploadedFile.type,
              lastModified: uploadedFile.lastModified,
              size: Math.round(uploadedFile.size / 2),
              loading: true,
            })),
          },
        });

        const documentCategories: IVariables = {
          thawtheeq: 'thawtheeq',
          noc: 'noc',
        };
        const promise = files.map(async (uploadedFile: any) => {
          const res = await InternalApi.uploadDedDoc(uploadedFile);
          const payload = {
            fieldName,
            id: res.data.uploadedFileDetails.nameOfFile,
            lastModifiedDate: uploadedFile.lastModifiedDate,
            name: uploadedFile.name,
            documentName: uploadedFile.name,
            documentCategory: documentCategories[fieldName],
            documentPath: res.data.uploadedFileDetails.s3FilePath,
            type: uploadedFile.type,
            lastModified: uploadedFile.lastModified,
            size: uploadedFile.size,
            loading: false,
          };
          return payload;
        });

        const resDocuments = await Promise.all(promise);

        handleChange({
          documents: {
            ...documents,
            [fieldName]: resDocuments,
          },
        });
      } catch (err) {
        handleChange({
          documents: {
            ...documents,
            [fieldName]: null,
          },
        });
        console.info('Error while uploading document:', err);
      }
    };
  };

  const onFileRemove = (fieldName: string) => {
    return () => {
      handleChange({
        documents: {
          ...documents,
          [fieldName]: null,
        },
      });
    };
  };

  const isUploaded = (fieldName: string) => {
    const isUploadedBool = has(documents, `${fieldName}[0].documentPath`);
    return isUploadedBool;
  };

  const getFiles = (fieldName: string) => {
    return documents[fieldName] === null || documents[fieldName] === undefined
      ? []
      : documents[fieldName].map((document: IVariables) => ({
          file: document,
          status: document.loading ? 'progress' : 'success',
          uploaded: document.loading ? 2.2 : document.size,
          loading: document.loading,
        }));
  };

  return (
    inputGroups && (
      <Form name="file-upload-section">
        {inputGroups.map((formGroups: IVariables) => {
          return (
            <Form.Fieldset title={i18n(formGroups.name)}>
              {formGroups.fields.map((fileConfig: IVariables) => {
                const { name, label, help, message, ...restProps } = fileConfig;
                if (name === 'thawtheeq' && props.isTawtheeqRequired)
                  return (
                    <FileUpload
                      i18n={i18n}
                      accept={['application/pdf']}
                      disabled={false}
                      help={
                        (startShowingErrors &&
                          !isUploaded(name) &&
                          i18n(help)) ||
                        (!isUploaded(name) && i18n(message)) ||
                        ''
                      }
                      label={i18n(label)}
                      maxSize={100002200}
                      multiple={false}
                      name={name}
                      validateStatus={
                        startShowingErrors && !isUploaded(name) ? 'error' : ''
                      }
                      {...restProps}
                      files={getFiles(name)}
                      onChange={onFileUpload(name)}
                      onRemove={onFileRemove(name)}
                    />
                  );
                return null;
              })}
            </Form.Fieldset>
          );
        })}
      </Form>
    )
  );
}
export default FileUploads;
