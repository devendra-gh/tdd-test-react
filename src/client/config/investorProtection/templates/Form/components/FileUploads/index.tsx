import React, { useState } from 'react';
import Form from '@tamm/ui-lib-v2-form';
import { IVariables } from '@tamm/app-composer';
import FileUpload from '@tamm/ui-lib-v2-file-upload';
import InternalApi from 'client/services/InternalApi';
import './template.less';

/* istanbul ignore file */

const MAX_FILE_SIZE = 1024 * 1024 * 4;

function FileUploads(props: IVariables) {
  const {
    inputGroups,
    handleChange,
    values,
    i18n,
    startShowingError,
    name: stateKey,
  } = props;
  const { [stateKey]: documents } = values;
  const [isUploading, setIsUploading] = useState(false);
  const handleFileChange = (fieldName: string, files: IVariables[] | null) =>
    handleChange({
      [stateKey]: {
        ...documents,
        [fieldName]: files,
      },
    });
  const onFileUpload = (fieldName: string) => {
    const previousFiles = documents[fieldName] || [];
    return async (files: IVariables) => {
      // 4MB validation
      if (files[0].size > MAX_FILE_SIZE) {
        console.info(i18n('validationMessage.file.size'));
        return;
      }
      setIsUploading(true);
      try {
        handleFileChange(
          fieldName,
          previousFiles
            .concat(
              files.map((uploadedFile: any) => ({
                fieldName,
                lastModifiedDate: uploadedFile.lastModifiedDate,
                name: uploadedFile.name,
                documentName: uploadedFile.name,
                type: uploadedFile.type,
                lastModified: uploadedFile.lastModified,
                size: Math.round(uploadedFile.size / 2),
                loading: true,
              })),
            )
            .map((file: IVariables, index: number) => ({
              ...file,
              id: index,
            })),
        );
        const promise = files.map(async (uploadedFile: any) => {
          const res = await InternalApi.uploadDedDoc(uploadedFile);
          const payload = {
            fieldName,
            id: res.data.uploadedFileDetails.nameOfFile,
            lastModifiedDate: uploadedFile.lastModifiedDate,
            name: uploadedFile.name,
            documentName: uploadedFile.name,
            documentPath: res.data.uploadedFileDetails.s3FilePath,
            type: uploadedFile.type,
            lastModified: uploadedFile.lastModified,
            size: uploadedFile.size,
            loading: false,
          };
          return payload;
        });
        const result = await Promise.all(promise);
        const currentDocuments = documents[fieldName] || [];
        const mergedArray = [...result, ...currentDocuments];
        // mergedArray have duplicates, lets remove the duplicates using Set
        const set = new Set();
        const finalDocuments = mergedArray.filter(item => {
          if (!set.has(item.name)) {
            set.add(item.name);
            return true;
          }
          return false;
        }, set);
        handleFileChange(fieldName, finalDocuments);
      } catch (err) {
        console.error('Error while uploading document:', err);
      } finally {
        setIsUploading(false);
      }
    };
  };
  const onFileRemove = (fieldName: string) => {
    return (file: IVariables) => {
      const updatedFiles: any = documents[fieldName].filter(
        (stateFile: IVariables) => stateFile.id !== file.id,
      );
      handleFileChange(fieldName, updatedFiles);
    };
  };
  const getFiles = (fieldName: string) => {
    const documentList = documents[fieldName];
    return documentList === null || documentList === undefined
      ? []
      : documentList.map((document: IVariables) => ({
          ...document,
          file: { ...document, type: 'custom' },
          status: document.loading ? 'progress' : 'success',
          uploaded: document.loading ? 2.2 : document.size,
          loading: document.loading,
          type: 'custom',
          size: document.size || '',
        }));
  };
  const fileValid = (fieldName: string, validationConfig: string) => {
    if (
      validationConfig === 'required' &&
      startShowingError &&
      (!values.documents[fieldName] || values.documents[fieldName].length === 0)
    ) {
      return true;
    }
    return false;
  };
  return (
    <div className="investor-form">
      {inputGroups && (
        <Form name={props.name} id={props.name}>
          {inputGroups.map((formGroups: IVariables) => (
            <Form.Fieldset
              title={i18n(formGroups.name)}
              key={`${props.groupKey}.${props.stateKey}`}
            >
              {formGroups.fields.map((fileConfig: IVariables) => {
                const {
                  name,
                  label,
                  help,
                  validationConfig,
                  ...restProps
                } = fileConfig;
                const files = getFiles(name);
                const helpString = i18n(help);

                /* if (
                  values.documents[name] && !(values.documents[name].length === 0)
                ) {
                  helpString = '';
                } */

                return (
                  <FileUpload
                    i18n={i18n}
                    accept={['application/pdf']}
                    multiple={props.multiple || false}
                    label={i18n(label)}
                    maxSize={4194304}
                    name={name}
                    {...restProps}
                    validateStatus={
                      fileValid(name, validationConfig) ? 'error' : ''
                    }
                    files={files}
                    help={helpString}
                    disabled={isUploading}
                    onChange={onFileUpload(name)}
                    onRemove={onFileRemove(name)}
                  />
                );
              })}
            </Form.Fieldset>
          ))}
        </Form>
      )}
    </div>
  );
}

export default FileUploads;
