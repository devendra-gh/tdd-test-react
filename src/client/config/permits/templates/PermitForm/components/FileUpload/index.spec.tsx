import React from 'react';
import { shallow } from 'enzyme';
import { IVariables } from '@tamm/app-composer';

// import { checkValidationField } from 'client/config/utils/checkValidation';
import FileUploads from './index';

jest.mock('client/config/utils/checkValidation', () => ({
  checkValidationField: jest
    .fn()
    .mockReturnValueOnce('true')
    .mockReturnValueOnce(''),
}));

jest.mock('client/services/InternalApi', () => ({
  uploadDedDoc: jest
    .fn()
    .mockReturnValueOnce(
      Promise.resolve({
        data: {
          uploadedFileDetails: {
            nameOfFile: 'name',
            s3FilePath: 'https://test-s3-url',
          },
        },
      }),
    )
    .mockReturnValueOnce(Promise.reject(new Error('error'))),
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Permit Type form', () => {
  let props: any;
  // const mockValidation: any = checkValidationField;
  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      serviceType: null,
      onSubmit: jest.fn(),
      formFields: {
        testField: {
          name: 'test field',
          stateKey: 'testStateKey',
          fields: [
            {
              validationConfig: {},
              elementType: 'input',
              onChange: () => jest.fn(),
            },
          ],
        },
      },
      values: {
        documents: {
          test: [
            {
              lastModifiedDate: '',
              lastModified: '',
              name: 'test',
              type: '',
              loading: true,
              size: '',
              value: 'dummy',
              documentPath: '',
            },
          ],
        },
      },
      inputGroups: [
        {
          name: 'test',
          fields: [
            {
              validationConfig: {
                type: 'REQUIRES FILE',
              },
              elementType: 'input',
              name: 'test',
              onChange: () => jest.fn(),
            },
          ],
        },
      ],
      handleChange: jest.fn(),
      startShowingErrors: true,
      name: 'documents',
    };
  });

  it('should successfully render when serviceType is provided', () => {
    props.serviceType = 'testServiceType';
    const component = shallow(<FileUploads {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should successfully render when all else statement', () => {
    props.inputGroups = [
      {
        name: 'test',
        fields: [
          {
            validationConfig: {},
            elementType: 'input',
            onChange: () => jest.fn(),
          },
        ],
      },
    ];

    const component = shallow(<FileUploads {...props} />);
    expect(component).toMatchSnapshot();
  });

  // it('should successfully upload file', () => {
  //   const component = shallow(<FileUploads {...props} />);
  //   const FileUpload = component.find('Memo(FileUpload)');
  //   const files: IVariables[] = [
  //     {
  //       fieldName: 'test field',
  //       lastModifiedDate: 'test date',
  //       name: 'test name',
  //       type: 'custom',
  //       lastModified: 'test last modified',
  //       size: 200,
  //     },
  //   ];
  //   FileUpload.simulate('change', files);
  //   expect(props.handleChange).toHaveBeenCalled();
  // });

  it('should successfully remove file', async () => {
    const component = shallow(<FileUploads {...props} />);
    const FileUpload = component.find('FileUpload');
    const files: IVariables[] = [
      {
        fieldName: 'test field',
        lastModifiedDate: 'test date',
        name: 'test name',
        type: 'custom',
        lastModified: 'test last modified',
        size: 200,
      },
    ];
    await FileUpload.simulate('change', files);
    await FileUpload.simulate('remove', 'test name');
    expect(props.handleChange).toHaveBeenCalled();
  });
});
