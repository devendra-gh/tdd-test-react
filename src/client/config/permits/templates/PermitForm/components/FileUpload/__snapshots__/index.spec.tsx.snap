// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permit Type form should successfully render when all else statement 1`] = `
<LocalizedComponent(Form)
  backButton={null}
  cancelLink={null}
  help=""
  i18n={[Function]}
  id="documents"
  name="documents"
  submitButton={null}
  validateStatus={null}
>
  <Form.Fieldset
    buttonGroup={false}
    gapSize="default"
    key="undefined.undefined"
    title=""
    twoColumns={false}
  >
    <Row
      className=""
      flex={true}
      gutter={20}
      style={null}
    >
      <Col
        className=""
        key="undefined.undefined"
        lg={null}
        md={4}
        offset={0}
        order={0}
        pull={0}
        push={0}
        sm={null}
        span={0}
        style={null}
        xl={4}
        xs={12}
      >
        <FileUpload
          accept={
            Array [
              "application/pdf",
            ]
          }
          disabled={false}
          elementType="input"
          files={Array []}
          help=""
          i18n={
            [MockFunction] {
              "calls": Array [
                Array [
                  "test",
                ],
                Array [
                  undefined,
                ],
                Array [
                  undefined,
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          maxSize={100002200}
          multiple={false}
          onChange={[Function]}
          onRemove={[Function]}
          validateStatus=""
        />
      </Col>
    </Row>
  </Form.Fieldset>
</LocalizedComponent(Form)>
`;

exports[`Permit Type form should successfully render when serviceType is provided 1`] = `
<LocalizedComponent(Form)
  backButton={null}
  cancelLink={null}
  help=""
  i18n={[Function]}
  id="documents"
  name="documents"
  submitButton={null}
  validateStatus={null}
>
  <Form.Fieldset
    buttonGroup={false}
    gapSize="default"
    key="undefined.undefined"
    title=""
    twoColumns={false}
  >
    <Row
      className=""
      flex={true}
      gutter={20}
      style={null}
    >
      <Col
        className=""
        key="test.undefined"
        lg={null}
        md={4}
        offset={0}
        order={0}
        pull={0}
        push={0}
        sm={null}
        span={0}
        style={null}
        xl={4}
        xs={12}
      >
        <FileUpload
          accept={
            Array [
              "application/pdf",
            ]
          }
          disabled={false}
          elementType="input"
          files={
            Array [
              Object {
                "documentPath": "",
                "file": Object {
                  "documentPath": "",
                  "lastModified": "",
                  "lastModifiedDate": "",
                  "loading": true,
                  "name": "test",
                  "size": "",
                  "type": "custom",
                  "value": "dummy",
                },
                "lastModified": "",
                "lastModifiedDate": "",
                "loading": true,
                "name": "test",
                "size": "",
                "status": "progress",
                "type": "custom",
                "uploaded": 2.2,
                "value": "dummy",
              },
            ]
          }
          help=""
          i18n={
            [MockFunction] {
              "calls": Array [
                Array [
                  "test",
                ],
                Array [
                  undefined,
                ],
                Array [
                  undefined,
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          maxSize={100002200}
          multiple={false}
          name="test"
          onChange={[Function]}
          onRemove={[Function]}
          validateStatus=""
        />
      </Col>
    </Row>
  </Form.Fieldset>
</LocalizedComponent(Form)>
`;
